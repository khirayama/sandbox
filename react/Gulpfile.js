var gulp = require('gulp');
var gReact = require('gulp-react');
var gReplace = require('gulp-replace');
var browserify = require('browserify');
var source = require('vinyl-source-stream');
var del = require('del');
var runSequence = require('run-sequence');

var browserifyConfig = {
  entries: ['./index.js'],
  standalone: 'Flux'
};

gulp.task('clean', function(cb) {
  del(['lib/', 'Flux.js'], cb);
});

gulp.task('lib', function() {
  return gulp.src('src/*.js')
             .pipe(gReact({harmony: true}))
             .pipe(gReplace(/__DEV__/g, 'false'))
             .pipe(gulp.dest('lib'));

});

gulp.task('browserify', ['lib'], function() {
  return browserify(browserifyConfig)
          .bundle()
          .pipe(source('Flux.js'))
          .pipe(gulp.dest('./dist/'))
});

gulp.task('build', ['lib', 'browserify']);

gulp.task('publish', function(cb) {
  runSequence('clean', 'build', cb);
});

gulp.task('default', ['build']);
