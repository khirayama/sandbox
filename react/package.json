{
  "name": "flux",
  "version": "2.0.2",
  "description": "An application architecture based on a unidirectional data flow",
  "keywords": [
    "flux",
    "react",
    "facebook",
    "dispatcher"
  ],
  "homepage": "http://facebook.github.io/flux/",
  "bugs": "https://github.com/facebook/flux/issues",
  "files": [
    "index.js",
    "lib/",
    "LICENSE",
    "PATENTS"
  ],
  "main": "index.js",
  "scripts": {
    "prepublish": "gulp publish",
    "test": "jest"
  },
  "jest": {
    "rootDir": "src",
    "scriptPreprocessor": "../jest-preprocessor.js"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/facebook/flux"
  },
  "author": "Facebook",
  "contributors": [
    "Jing Chen <jingc@fb.com>",
    "Bill Fisher <fisherwebdev@gmail.com>",
    "Paul O'Shannessy <paul@oshanessy.com>"
  ],
  "license": "BSD",
  "devDependencies": {
    "browserify": "^9.0.3",
    "del": "^1.1.1",
    "gulp": "^3.8.11",
    "gulp-clean": "^0.3.1",
    "gulp-react": "^2.0.0",
    "gulp-replace": "^0.5.3",
    "jest-cli": "^0.4.0",
    "react-tools": "^0.12.0",
    "run-sequence": "^1.0.2",
    "vinyl-source-stream": "^1.0.0"
  }
}
