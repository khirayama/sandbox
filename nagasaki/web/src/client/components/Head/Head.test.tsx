import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Head } from '.';

test('should render self and sub-components', () => {
  const tree = shallow(<Head title="foo" />);

  expect(toJson(tree)).toMatchSnapshot();
});
