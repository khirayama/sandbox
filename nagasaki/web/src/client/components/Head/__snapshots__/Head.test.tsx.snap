// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render self and sub-components 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <meta
    charSet="UTF-8"
  />
  <meta
    content="width=device-width, initial-scale=1.0"
    name="viewport"
  />
  <title>
    foo
  </title>
  <link
    href="https://fonts.googleapis.com/css?family=Muli"
    rel="stylesheet"
  />
</HelmetWrapper>
`;
